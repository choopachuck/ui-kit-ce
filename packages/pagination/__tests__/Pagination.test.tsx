import * as React from 'react'
import { render, fireEvent } from '@testing-library/react'
import {
  Pagination,
  PaginationPageButtonProps,
  PaginationNavigationButtonProps,
  PAGINATION_OVERFLOW_PAGES,
} from '../src'
import PageButton from './testComponents/PageButton'
import SelectedIndicator from './testComponents/SelectedIndicator'
import NavigationButton from './testComponents/NavigationButton'
import OverflowButton from './testComponents/OverflowButton'

const PageItemMock = ({
  onClick,
  children,
  className,
}: PaginationPageButtonProps) => (
  <button title="page" className={className} onClick={onClick}>
    {children}
  </button>
)
const NavigationPrevMock = ({
  onPageChange,
  children,
  disabled,
}: PaginationNavigationButtonProps) => (
  <button disabled={disabled} title="prev" onClick={() => onPageChange?.()}>
    {children}
  </button>
)
const NavigationNextMock = ({
  onPageChange,
  children,
  disabled,
}: PaginationNavigationButtonProps) => (
  <button disabled={disabled} title="next" onClick={() => onPageChange?.()}>
    {children}
  </button>
)

const NavigationFirstMock = ({
  onPageChange,
  children,
  disabled,
}: PaginationNavigationButtonProps) => (
  <button disabled={disabled} title="first" onClick={() => onPageChange?.()}>
    {children}
  </button>
)
const NavigationLastMock = ({
  onPageChange,
  children,
  disabled,
}: PaginationNavigationButtonProps) => (
  <button disabled={disabled} title="last" onClick={() => onPageChange?.()}>
    {children}
  </button>
)

const itemSelectedClassName = 'selected'

jest.mock('./testComponents/PageButton', () => {
  return jest.fn(() => null)
})

jest.mock('./testComponents/SelectedIndicator', () => {
  return jest.fn(() => null)
})

jest.mock('./testComponents/NavigationButton', () => {
  return jest.fn(() => null)
})

jest.mock('./testComponents/OverflowButton', () => {
  return jest.fn(() => null)
})

it('elements count depends on totalPageCount property', () => {
  const { getAllByTitle, getAllByRole, rerender } = render(
    <Pagination totalPageCount={1} components={{ PageButton: PageItemMock }} />
  )

  expect(getAllByTitle('page').length).toEqual(1)

  rerender(
    <Pagination totalPageCount={2} components={{ PageButton: PageItemMock }} />
  )

  expect(getAllByTitle('page').length).toEqual(2)

  rerender(
    <Pagination totalPageCount={3} components={{ PageButton: PageItemMock }} />
  )

  expect(getAllByTitle('page').length).toEqual(3)

  rerender(<Pagination components={{ PageButton: PageItemMock }} />)

  expect(getAllByTitle('page').length).toEqual(1)

  rerender(
    <Pagination totalPageCount={0} components={{ PageButton: PageItemMock }} />
  )

  expect(getAllByRole('button').length).toEqual(2)
})

it('page change works correctly by click on page number', () => {
  const handlePageChange = jest.fn()
  const { getByText } = render(
    <Pagination
      totalPageCount={5}
      classes={{ itemSelected: itemSelectedClassName }}
      onPageChange={handlePageChange}
    />
  )

  expect(getByText('1')).toHaveClass(itemSelectedClassName)
  fireEvent.click(getByText('2'))
  expect(getByText('2')).toHaveClass(itemSelectedClassName)
  expect(handlePageChange).toBeCalledTimes(1)
  expect(handlePageChange).toHaveBeenCalledWith(2)

  fireEvent.click(getByText('3'))
  expect(getByText('2')).not.toHaveClass(itemSelectedClassName)
  expect(getByText('3')).toHaveClass(itemSelectedClassName)
  expect(handlePageChange).toBeCalledTimes(2)
  expect(handlePageChange).toHaveBeenCalledWith(3)
})

it('page change works correctly by click on navigation prev and next', () => {
  const handlePageChange = jest.fn()
  const { getByText, getByTitle } = render(
    <Pagination
      showNavigationFirst
      showNavigationLast
      totalPageCount={3}
      classes={{ itemSelected: itemSelectedClassName }}
      components={{
        LastPageButton: NavigationLastMock,
        FirstPageButton: NavigationFirstMock,
        PreviousPageButton: NavigationPrevMock,
        NextPageButton: NavigationNextMock,
      }}
      onPageChange={handlePageChange}
    />
  )

  const prevNavigationButton = getByTitle('prev')
  const nextNavigationButton = getByTitle('next')
  const firstNavigationButton = getByTitle('first')
  const lastNavigationButton = getByTitle('last')

  expect(prevNavigationButton).toBeDisabled()
  expect(nextNavigationButton).not.toBeDisabled()
  expect(firstNavigationButton).toBeDisabled()
  expect(lastNavigationButton).not.toBeDisabled()

  fireEvent.click(nextNavigationButton)

  expect(getByText('2')).toHaveClass(itemSelectedClassName)
  expect(handlePageChange).toBeCalledTimes(1)
  expect(handlePageChange).toHaveBeenCalledWith(2)

  expect(prevNavigationButton).not.toBeDisabled()
  expect(nextNavigationButton).not.toBeDisabled()
  expect(firstNavigationButton).not.toBeDisabled()
  expect(lastNavigationButton).not.toBeDisabled()

  fireEvent.click(nextNavigationButton)

  expect(getByText('3')).toHaveClass(itemSelectedClassName)
  expect(handlePageChange).toBeCalledTimes(2)
  expect(handlePageChange).toHaveBeenCalledWith(3)

  expect(prevNavigationButton).not.toBeDisabled()
  expect(nextNavigationButton).toBeDisabled()
  expect(firstNavigationButton).not.toBeDisabled()
  expect(lastNavigationButton).toBeDisabled()

  fireEvent.click(prevNavigationButton)

  expect(getByText('2')).toHaveClass(itemSelectedClassName)
  expect(handlePageChange).toBeCalledTimes(3)
  expect(handlePageChange).toHaveBeenCalledWith(2)

  expect(prevNavigationButton).not.toBeDisabled()
  expect(nextNavigationButton).not.toBeDisabled()
  expect(firstNavigationButton).not.toBeDisabled()
  expect(lastNavigationButton).not.toBeDisabled()

  fireEvent.click(prevNavigationButton)

  expect(getByText('1')).toHaveClass(itemSelectedClassName)
  expect(handlePageChange).toBeCalledTimes(4)
  expect(handlePageChange).toHaveBeenCalledWith(1)

  expect(prevNavigationButton).toBeDisabled()
  expect(nextNavigationButton).not.toBeDisabled()
  expect(firstNavigationButton).toBeDisabled()
  expect(lastNavigationButton).not.toBeDisabled()
})

it('page change works correctly by click on navigation first and last', () => {
  const handlePageChange = jest.fn()

  const { getByText, getByTitle } = render(
    <Pagination
      showNavigationFirst
      showNavigationLast
      totalPageCount={3}
      classes={{ itemSelected: itemSelectedClassName }}
      components={{
        LastPageButton: NavigationLastMock,
        FirstPageButton: NavigationFirstMock,
        PreviousPageButton: NavigationPrevMock,
        NextPageButton: NavigationNextMock,
      }}
      onPageChange={handlePageChange}
    />
  )

  const firstNavigationButton = getByTitle('first')
  const lastNavigationButton = getByTitle('last')

  const prevNavigationButton = getByTitle('prev')
  const nextNavigationButton = getByTitle('next')

  expect(firstNavigationButton).toBeDisabled()
  expect(lastNavigationButton).not.toBeDisabled()
  expect(prevNavigationButton).toBeDisabled()
  expect(nextNavigationButton).not.toBeDisabled()

  fireEvent.click(lastNavigationButton)

  expect(getByText('3')).toHaveClass(itemSelectedClassName)
  expect(handlePageChange).toBeCalledTimes(1)
  expect(handlePageChange).toHaveBeenCalledWith(3)

  expect(firstNavigationButton).not.toBeDisabled()
  expect(lastNavigationButton).toBeDisabled()
  expect(prevNavigationButton).not.toBeDisabled()
  expect(nextNavigationButton).toBeDisabled()

  fireEvent.click(firstNavigationButton)

  expect(getByText('1')).toHaveClass(itemSelectedClassName)
  expect(handlePageChange).toBeCalledTimes(2)
  expect(handlePageChange).toHaveBeenCalledWith(1)

  expect(firstNavigationButton).toBeDisabled()
  expect(lastNavigationButton).not.toBeDisabled()
  expect(prevNavigationButton).toBeDisabled()
  expect(nextNavigationButton).not.toBeDisabled()
})

it('default page renders correctly', () => {
  const { getByText } = render(
    <Pagination
      totalPageCount={5}
      defaultPage={3}
      classes={{ itemSelected: itemSelectedClassName }}
    />
  )

  expect(getByText('3')).toHaveClass(itemSelectedClassName)
})

it('near page count renders correctly', () => {
  const { getAllByRole } = render(
    <Pagination
      totalPageCount={10}
      classes={{ itemSelected: itemSelectedClassName }}
      nearPageCount={2}
    />
  )

  let items = getAllByRole('button')
  const nextNavigationButton = items[items.length - 1]

  expect(items[1]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual('2')
  expect(items[3].textContent).toEqual('3')
  expect(items[4].textContent).toEqual('4')
  expect(items[5].textContent).toEqual('5')
  expect(items[6].textContent).toEqual('6')
  expect(items[7].textContent).toEqual('7')
  expect(items[8].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[2]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual('2')
  expect(items[3].textContent).toEqual('3')
  expect(items[4].textContent).toEqual('4')
  expect(items[5].textContent).toEqual('5')
  expect(items[6].textContent).toEqual('6')
  expect(items[7].textContent).toEqual('7')
  expect(items[8].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[3]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual('2')
  expect(items[3].textContent).toEqual('3')
  expect(items[4].textContent).toEqual('4')
  expect(items[5].textContent).toEqual('5')
  expect(items[6].textContent).toEqual('6')
  expect(items[7].textContent).toEqual('7')
  expect(items[8].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[4]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual('2')
  expect(items[3].textContent).toEqual('3')
  expect(items[4].textContent).toEqual('4')
  expect(items[5].textContent).toEqual('5')
  expect(items[6].textContent).toEqual('6')
  expect(items[7].textContent).toEqual('7')
  expect(items[8].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[5]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[3].textContent).toEqual('3')
  expect(items[4].textContent).toEqual('4')
  expect(items[5].textContent).toEqual('5')
  expect(items[6].textContent).toEqual('6')
  expect(items[7].textContent).toEqual('7')
  expect(items[8].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[5]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[3].textContent).toEqual('4')
  expect(items[4].textContent).toEqual('5')
  expect(items[5].textContent).toEqual('6')
  expect(items[6].textContent).toEqual('7')
  expect(items[7].textContent).toEqual('8')
  expect(items[8].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[6]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[3].textContent).toEqual('4')
  expect(items[4].textContent).toEqual('5')
  expect(items[5].textContent).toEqual('6')
  expect(items[6].textContent).toEqual('7')
  expect(items[7].textContent).toEqual('8')
  expect(items[8].textContent).toEqual('9')
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[7]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[3].textContent).toEqual('4')
  expect(items[4].textContent).toEqual('5')
  expect(items[5].textContent).toEqual('6')
  expect(items[6].textContent).toEqual('7')
  expect(items[7].textContent).toEqual('8')
  expect(items[8].textContent).toEqual('9')
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[8]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[3].textContent).toEqual('4')
  expect(items[4].textContent).toEqual('5')
  expect(items[5].textContent).toEqual('6')
  expect(items[6].textContent).toEqual('7')
  expect(items[7].textContent).toEqual('8')
  expect(items[8].textContent).toEqual('9')
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[9]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[3].textContent).toEqual('4')
  expect(items[4].textContent).toEqual('5')
  expect(items[5].textContent).toEqual('6')
  expect(items[6].textContent).toEqual('7')
  expect(items[7].textContent).toEqual('8')
  expect(items[8].textContent).toEqual('9')
  expect(items[9].textContent).toEqual('10')
})

it('boundary page count renders correctly', () => {
  const { getAllByRole } = render(
    <Pagination
      totalPageCount={10}
      classes={{ itemSelected: itemSelectedClassName }}
      boundaryPageCount={2}
    />
  )

  let items = getAllByRole('button')
  const nextNavigationButton = items[items.length - 1]

  expect(items[1]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual('2')
  expect(items[3].textContent).toEqual('3')
  expect(items[4].textContent).toEqual('4')
  expect(items[5].textContent).toEqual('5')
  expect(items[6].textContent).toEqual('6')
  expect(items[7].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[8].textContent).toEqual('9')
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[2]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual('2')
  expect(items[3].textContent).toEqual('3')
  expect(items[4].textContent).toEqual('4')
  expect(items[5].textContent).toEqual('5')
  expect(items[6].textContent).toEqual('6')
  expect(items[7].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[8].textContent).toEqual('9')
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[3]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual('2')
  expect(items[3].textContent).toEqual('3')
  expect(items[4].textContent).toEqual('4')
  expect(items[5].textContent).toEqual('5')
  expect(items[6].textContent).toEqual('6')
  expect(items[7].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[8].textContent).toEqual('9')
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[4]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual('2')
  expect(items[3].textContent).toEqual('3')
  expect(items[4].textContent).toEqual('4')
  expect(items[5].textContent).toEqual('5')
  expect(items[6].textContent).toEqual('6')
  expect(items[7].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[8].textContent).toEqual('9')
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[5]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual('2')
  expect(items[3].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[4].textContent).toEqual('4')
  expect(items[5].textContent).toEqual('5')
  expect(items[6].textContent).toEqual('6')
  expect(items[7].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[8].textContent).toEqual('9')
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[5]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual('2')
  expect(items[3].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[4].textContent).toEqual('5')
  expect(items[5].textContent).toEqual('6')
  expect(items[6].textContent).toEqual('7')
  expect(items[7].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[8].textContent).toEqual('9')
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[6]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual('2')
  expect(items[3].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[4].textContent).toEqual('5')
  expect(items[5].textContent).toEqual('6')
  expect(items[6].textContent).toEqual('7')
  expect(items[7].textContent).toEqual('8')
  expect(items[8].textContent).toEqual('9')
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[7]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual('2')
  expect(items[3].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[4].textContent).toEqual('5')
  expect(items[5].textContent).toEqual('6')
  expect(items[6].textContent).toEqual('7')
  expect(items[7].textContent).toEqual('8')
  expect(items[8].textContent).toEqual('9')
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[8]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual('2')
  expect(items[3].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[4].textContent).toEqual('5')
  expect(items[5].textContent).toEqual('6')
  expect(items[6].textContent).toEqual('7')
  expect(items[7].textContent).toEqual('8')
  expect(items[8].textContent).toEqual('9')
  expect(items[9].textContent).toEqual('10')

  fireEvent.click(nextNavigationButton)
  items = getAllByRole('button')
  expect(items[9]).toHaveClass(itemSelectedClassName)

  expect(items[1].textContent).toEqual('1')
  expect(items[2].textContent).toEqual('2')
  expect(items[3].textContent).toEqual(PAGINATION_OVERFLOW_PAGES)
  expect(items[4].textContent).toEqual('5')
  expect(items[5].textContent).toEqual('6')
  expect(items[6].textContent).toEqual('7')
  expect(items[7].textContent).toEqual('8')
  expect(items[8].textContent).toEqual('9')
  expect(items[9].textContent).toEqual('10')
})

it('pagination components renders page component correctly', () => {
  render(<Pagination totalPageCount={10} components={{ PageButton }} />)

  expect(PageButton).toBeCalledTimes(6)
  expect(PageButton).toMatchSnapshot()
})

it('pagination components renders navigation first component correctly', () => {
  render(
    <Pagination
      showNavigationFirst
      totalPageCount={10}
      components={{ FirstPageButton: NavigationButton }}
    />
  )

  expect(NavigationButton).toMatchSnapshot()
})

it('pagination components renders navigation last component correctly', () => {
  render(
    <Pagination
      showNavigationLast
      totalPageCount={10}
      components={{ LastPageButton: NavigationButton }}
    />
  )

  expect(NavigationButton).toMatchSnapshot()
})

it('pagination components renders navigation previous component correctly', () => {
  render(
    <Pagination
      totalPageCount={10}
      components={{ PreviousPageButton: NavigationButton }}
    />
  )

  expect(NavigationButton).toMatchSnapshot()
})

it('pagination components renders navigation next component correctly', () => {
  render(
    <Pagination
      totalPageCount={10}
      components={{ NextPageButton: NavigationButton }}
    />
  )

  expect(NavigationButton).toMatchSnapshot()
})

it('pagination components renders divider component correctly', () => {
  render(<Pagination totalPageCount={10} components={{ SelectedIndicator }} />)

  expect(SelectedIndicator).toMatchSnapshot()
})

it('pagination components renders overflow component correctly', () => {
  render(
    <Pagination
      totalPageCount={10}
      currentPage={5}
      components={{ OverflowButton }}
    />
  )

  expect(OverflowButton).toMatchSnapshot()
})
