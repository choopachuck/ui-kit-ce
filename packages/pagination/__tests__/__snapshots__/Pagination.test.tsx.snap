// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination components renders divider component correctly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "className": "pagination-selectedIndicator-0-2-8 pagination-selectedIndicatorMedium-0-2-10",
      },
      Object {},
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": null,
    },
  ],
}
`;

exports[`pagination components renders navigation first component correctly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "aria-label": "Go to first page",
        "children": <Unknown />,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 pagination-navigationButton-0-2-12 pagination-navigationButtonFirst-0-2-15 pagination-itemDisabled-0-2-17",
        "disabled": true,
        "isEnd": false,
        "isStart": true,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": null,
    },
  ],
}
`;

exports[`pagination components renders navigation last component correctly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "aria-label": "Go to first page",
        "children": <Unknown />,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 pagination-navigationButton-0-2-12 pagination-navigationButtonFirst-0-2-15 pagination-itemDisabled-0-2-17",
        "disabled": true,
        "isEnd": false,
        "isStart": true,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
    Array [
      Object {
        "aria-label": "Go to last page",
        "children": <Unknown />,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 pagination-navigationButton-0-2-12 pagination-navigationButtonLast-0-2-16",
        "disabled": false,
        "isEnd": false,
        "isStart": true,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": null,
    },
    Object {
      "type": "return",
      "value": null,
    },
  ],
}
`;

exports[`pagination components renders navigation next component correctly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "aria-label": "Go to first page",
        "children": <Unknown />,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 pagination-navigationButton-0-2-12 pagination-navigationButtonFirst-0-2-15 pagination-itemDisabled-0-2-17",
        "disabled": true,
        "isEnd": false,
        "isStart": true,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
    Array [
      Object {
        "aria-label": "Go to last page",
        "children": <Unknown />,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 pagination-navigationButton-0-2-12 pagination-navigationButtonLast-0-2-16",
        "disabled": false,
        "isEnd": false,
        "isStart": true,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
    Array [
      Object {
        "aria-label": "Go to previous page",
        "children": <Unknown />,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 pagination-navigationButton-0-2-12 pagination-navigationButtonPrevious-0-2-13 pagination-itemDisabled-0-2-17",
        "disabled": true,
        "isEnd": false,
        "isStart": true,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
    Array [
      Object {
        "aria-label": "Go to next page",
        "children": <Unknown />,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 pagination-navigationButton-0-2-12 pagination-navigationButtonNext-0-2-14",
        "disabled": false,
        "isEnd": false,
        "isStart": true,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": null,
    },
    Object {
      "type": "return",
      "value": null,
    },
    Object {
      "type": "return",
      "value": null,
    },
    Object {
      "type": "return",
      "value": null,
    },
  ],
}
`;

exports[`pagination components renders navigation previous component correctly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "aria-label": "Go to first page",
        "children": <Unknown />,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 pagination-navigationButton-0-2-12 pagination-navigationButtonFirst-0-2-15 pagination-itemDisabled-0-2-17",
        "disabled": true,
        "isEnd": false,
        "isStart": true,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
    Array [
      Object {
        "aria-label": "Go to last page",
        "children": <Unknown />,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 pagination-navigationButton-0-2-12 pagination-navigationButtonLast-0-2-16",
        "disabled": false,
        "isEnd": false,
        "isStart": true,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
    Array [
      Object {
        "aria-label": "Go to previous page",
        "children": <Unknown />,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 pagination-navigationButton-0-2-12 pagination-navigationButtonPrevious-0-2-13 pagination-itemDisabled-0-2-17",
        "disabled": true,
        "isEnd": false,
        "isStart": true,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": null,
    },
    Object {
      "type": "return",
      "value": null,
    },
    Object {
      "type": "return",
      "value": null,
    },
  ],
}
`;

exports[`pagination components renders overflow component correctly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "appearance": "overflow-start",
        "children": "...",
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 pagination-itemOverflow-0-2-7",
        "disabled": false,
        "size": "md",
      },
      Object {},
    ],
    Array [
      Object {
        "appearance": "overflow-end",
        "children": "...",
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 pagination-itemOverflow-0-2-7",
        "disabled": false,
        "size": "md",
      },
      Object {},
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": null,
    },
    Object {
      "type": "return",
      "value": null,
    },
  ],
}
`;

exports[`pagination components renders page component correctly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "appearance": "item",
        "aria-label": "Go to page 1",
        "children": 1,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 resetButton-0-2-19 pagination-item-0-2-2 pagination-itemSelected-0-2-6",
        "disabled": false,
        "indicator": <ForwardRef
          className="pagination-selectedIndicator-0-2-8 pagination-selectedIndicatorMedium-0-2-10"
        />,
        "isSelected": true,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
    Array [
      Object {
        "appearance": "item",
        "aria-label": "Go to page 2",
        "children": 2,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 resetButton-0-2-19 pagination-item-0-2-2",
        "disabled": false,
        "indicator": <ForwardRef
          className="pagination-selectedIndicator-0-2-8 pagination-selectedIndicatorMedium-0-2-10"
        />,
        "isSelected": false,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
    Array [
      Object {
        "appearance": "item",
        "aria-label": "Go to page 3",
        "children": 3,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 resetButton-0-2-19 pagination-item-0-2-2",
        "disabled": false,
        "indicator": <ForwardRef
          className="pagination-selectedIndicator-0-2-8 pagination-selectedIndicatorMedium-0-2-10"
        />,
        "isSelected": false,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
    Array [
      Object {
        "appearance": "item",
        "aria-label": "Go to page 4",
        "children": 4,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 resetButton-0-2-19 pagination-item-0-2-2",
        "disabled": false,
        "indicator": <ForwardRef
          className="pagination-selectedIndicator-0-2-8 pagination-selectedIndicatorMedium-0-2-10"
        />,
        "isSelected": false,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
    Array [
      Object {
        "appearance": "item",
        "aria-label": "Go to page 5",
        "children": 5,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 resetButton-0-2-19 pagination-item-0-2-2",
        "disabled": false,
        "indicator": <ForwardRef
          className="pagination-selectedIndicator-0-2-8 pagination-selectedIndicatorMedium-0-2-10"
        />,
        "isSelected": false,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
    Array [
      Object {
        "appearance": "item",
        "aria-label": "Go to page 10",
        "children": 10,
        "className": "resetButton-0-2-19 pagination-item-0-2-2 pagination-itemMedium-0-2-4 resetButton-0-2-19 pagination-item-0-2-2",
        "disabled": false,
        "indicator": <ForwardRef
          className="pagination-selectedIndicator-0-2-8 pagination-selectedIndicatorMedium-0-2-10"
        />,
        "isSelected": false,
        "onPageChange": [Function],
        "size": "md",
      },
      Object {},
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": null,
    },
    Object {
      "type": "return",
      "value": null,
    },
    Object {
      "type": "return",
      "value": null,
    },
    Object {
      "type": "return",
      "value": null,
    },
    Object {
      "type": "return",
      "value": null,
    },
    Object {
      "type": "return",
      "value": null,
    },
  ],
}
`;
